From: Markus Koschany <apo@debian.org>
Date: Sun, 13 Dec 2015 16:28:22 +0100
Subject: python PIL tobytes error

Fix runtime error due to API changes in Python PIL.

Bug: https://bugs.debian.org/807250
Forwarded: no
---
 src/Texture.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/Texture.py b/src/Texture.py
index 51509f5..9b3b6de 100644
--- a/src/Texture.py
+++ b/src/Texture.py
@@ -210,13 +210,13 @@ class Texture:
     """Load the texture from a PIL image"""
     image = image.transpose(Image.FLIP_TOP_BOTTOM)
     if image.mode == "RGBA":
-      string = image.tostring('raw', 'RGBA', 0, -1)
+      string = image.tobytes('raw', 'RGBA', 0, -1)
       self.loadRaw(image.size, string, GL_RGBA, 4)
     elif image.mode == "RGB":
-      string = image.tostring('raw', 'RGB', 0, -1)
+      string = image.tobytes('raw', 'RGB', 0, -1)
       self.loadRaw(image.size, string, GL_RGB, 3)
     elif image.mode == "L":
-      string = image.tostring('raw', 'L', 0, -1)
+      string = image.tobytes('raw', 'L', 0, -1)
       self.loadRaw(image.size, string, GL_LUMINANCE, 1)
     else:
       raise TextureException("Unsupported image mode '%s'" % image.mode)
@@ -256,7 +256,7 @@ class Texture:
       # appears to be using PIL to do the conversion.
       string = pygame.image.tostring(surface, "RGB")
       image = Image.fromstring("RGB", surface.get_size(), string).convert("L")
-      string = image.tostring('raw', 'L', 0, -1)
+      string = image.tobytes('raw', 'L', 0, -1)
       self.loadRaw(surface.get_size(), string, GL_LUMINANCE, GL_INTENSITY8)
     else:
       if alphaChannel:
@@ -275,7 +275,7 @@ class Texture:
       # appears to be using PIL to do the conversion.
       string = pygame.image.tostring(surface, "RGB")
       image = Image.fromstring("RGB", surface.get_size(), string).convert("L")
-      string = image.tostring('raw', 'L', 0, -1)
+      string = image.tobytes('raw', 'L', 0, -1)
       self.loadSubRaw(surface.get_size(), position, string, GL_INTENSITY8)
     else:
       if alphaChannel:
